cocos/particle/particle-system.ts

```diff
@@ -1109,6 +1109,7 @@ export class ParticleSystem extends ModelRenderer {
         if (this._trailModule) this._trailModule.onDisable();
         if (this._boundingBox) {
             this._boundingBox = null;
+            this._oldPos = null;
```

Comment:
Good catch.

But why put the logic inside `if (this._boundingBox)`? 

Should it be outside `if (this._boundingBox)` ?

Reply:
Mostly consistency, but looking over the code again, I can see cases where the boundBox has already been set to null before onDisable is called so probably safest to just always set it to null. Good call out.